src/CoreShop/Component/Core/Order/Processor/CartItemsWholesaleProcessor.php

```diff
@@ -22,38 +23,56 @@
 
 final class CartItemsWholesaleProcessor implements CartProcessorInterface
 {
+    /**
+     * @var CartContextResolverInterface
+     */
+    private $cartContextResolver;
+
     /**
      * @var PurchasableWholesalePriceCalculatorInterface
      */
     private $wholesalePriceCalculator;
 
     /**
      * @param PurchasableWholesalePriceCalculatorInterface $wholesalePriceCalculator
+     * @param CartContextResolverInterface                 $cartContextResolver
      */
-    public function __construct(PurchasableWholesalePriceCalculatorInterface $wholesalePriceCalculator)
-    {
+    public function __construct(
+        PurchasableWholesalePriceCalculatorInterface $wholesalePriceCalculator,
+        CartContextResolverInterface $cartContextResolver = null
+    ) {
         $this->wholesalePriceCalculator = $wholesalePriceCalculator;
+        $this->cartContextResolver = $cartContextResolver;
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(CartInterface $cart)
     {
-        $store = $cart->getStore();
+        if (null === $this->cartContextResolver) {
+            @trigger_error(
+                'Using CartItemsWholesaleProcessor without a CarContextResolverInterpreter is deprecated since 2.1.2 and will be removed with 3.0.0',
```

Comment:
```suggestion
                'Using CartItemsWholesaleProcessor without a CartContextResolverInterface is deprecated since 2.1.2 and will be removed with 3.0.0',
```

Reply:
done